base/combinatorics.jl

```diff
@@ -185,7 +184,7 @@ julia> A
  1
 ```
 """
-permute!(a, p::AbstractVector) = permute!!(a, copymutable(p))
+permute!(v, p::AbstractVector) = copyto!(v, v[p])
```

Comment:
The issue I have with this is that it seems to violate the basic expectation of `permute!` by allocating a copy of the array `v`.  If the user wanted `v .= v[p]`, why would they call `permute!(v, p)`?

What's the point of having this function at all if it doesn't actually work in-place in `v`?

Reply:
I don't think that `permute!(::AbstractVector, ::AbstractVector)` is a natural function to define. It is an inherently inefficient way to manipulate data and so I do not believe it belongs in base, but that's a 2.0 change.